zone/attack.cpp

```diff
@@ -2371,6 +2371,41 @@ void NPC::Damage(Mob* other, int64 damage, uint16 spell_id, EQ::skills::SkillTyp
 	}
 }
 
+bool NPC::ShouldLeaveCorpseUponDeath(Mob* killer_mob, Mob* killer, bool damageTopWasZero, bool isLdonTreasure) {
+    if (damageTopWasZero) {
+        return false;
+    }
+
+    bool allow_merchant_corpse = RuleB(Merchant, AllowCorpse);
+    bool is_merchant = (class_ == MERCHANT || class_ == ADVENTURE_MERCHANT || MerchantType != 0);
+
+    // Initial checks on current entity
+    bool hasNoOwner = !HasOwner();
```

Comment:
All of these locally scoped variables should be `snake_case` instead of `camelCase`